packages/aws-client/src/S3.ts

```diff
@@ -1,44 +1,49 @@
-const fs = require('fs');
-const isString = require('lodash/isString');
-const path = require('path');
-const pMap = require('p-map');
-const pRetry = require('p-retry');
-const pump = require('pump');
-const url = require('url');
-
-const {
+/**
+ * @module S3
+ */
+
+import fs = require('fs');
+import path = require('path');
+import pMap = require('p-map');
+import pRetry = require('p-retry');
+import pump = require('pump');
+import url = require('url');
+import { Readable, TransformOptions } from 'stream';
+
+import {
   generateChecksumFromStream,
   validateChecksumFromStream
-} = require('@cumulus/checksum');
-const {
+} from '@cumulus/checksum';
+import {
   InvalidChecksum,
   UnparsableFileLocationError
-} = require('@cumulus/errors');
-const Logger = require('@cumulus/logger');
+} from '@cumulus/errors';
+import Logger = require('@cumulus/logger');
 
-const awsServices = require('./services');
-const { inTestMode } = require('./test-utils');
-const { improveStackTrace } = require('./utils');
+import { s3 } from './services';
+import { inTestMode } from './test-utils';
+import { improveStackTrace } from './utils';
 
 const log = new Logger({ sender: 'aws-client/s3' });
 
-let S3_RATE_LIMIT = 20;
-if (inTestMode()) {
-  S3_RATE_LIMIT = 1;
-}
+const S3_RATE_LIMIT = inTestMode() ? 1 : 20;
 
 /**
  * Join strings into an S3 key without a leading slash or double slashes
  *
  * @param {...string|Array<string>} args - the strings to join
  * @returns {string} the full S3 key
+ *
+ * @static
```

Comment:
What does this mean outside of a class context?

Reply:
This falls under the "I don't always understand how `jsdoc2md` works" category. This was an issue in JS too, it isn't new in TS.

If you export a function that was declared as `const listS3Objects = () => blah;`, the `jsdoc2md` will generate documentation as expected.

If you export a function that was declared as `function s3Join () {}`, the `jsdoc2md` documents it as this, for some reason:

```
    * _inner_
        * [~s3Join(...args)](#module_S3..s3Join) ⇒ <code>string</code>
```

I don't know why, but I know that adding `@static` to the jsdoc of those `function blah()` exports will fix it.

Reply:
Any reason to not just export it as a `const` instead?

Reply:
Yeah, looks like that fixes it. I'll make that change.